src/internal/components/InternalEditor.tsx

```diff
@@ -97,41 +96,62 @@ export const InternalEditor = ({
   };
 
   const handleSave = async (data: Data) => {
+    if (themeModeActive) {
+      devLogger.logFunc("saveStyles");
+      // save styles
+      return;
+    }
     devLogger.logFunc("saveVisualConfigData");
     saveVisualConfigData({
       payload: { visualConfigurationData: JSON.stringify(data) },
     });
   };
 
-  const change = async () => {
-    if (isLoading) {
-      return;
-    }
-    if (!canEdit) {
-      setCanEdit(true);
-      return;
-    }
+  const toggleThemeModeActive = () => {
+    setThemeModeActive((prev) => !prev);
   };
 
+  // set permissions on the component level to allow for dynamic updating
+  Object.values(puckConfig.components).forEach((component) => {
+    component.resolvePermissions = () => {
+      return {
+        drag: !themeModeActive,
+        duplicate: !themeModeActive,
+        delete: !themeModeActive,
+        insert: !themeModeActive,
+        edit: !themeModeActive,
+      };
+    };
+  });
+
   return (
     <EntityFieldProvider>
       <Puck
         config={puckConfig}
         data={{}} // we use puckInitialHistory instead
         initialHistory={puckInitialHistory}
-        onChange={change}
         overrides={{
           header: () => {
-            const { appState } = usePuck();
+            const { appState, refreshPermissions } = usePuck();
+
+            useEffect(() => {
+              refreshPermissions();
+            }, [themeModeActive]);
+
             return customHeader(
               handleClearLocalChanges,
               handleHistoryChange,
               appState.data,
               puckInitialHistory?.histories[0].state.data, // used for clearing local changes - reset to first puck history
               handleSave,
-              templateMetadata.isDevMode && !templateMetadata.devOverride
+              templateMetadata.isDevMode && !templateMetadata.devOverride,
+              themeModeActive,
+              toggleThemeModeActive
             );
           },
+          actionBar: themeModeActive ? () => <></> : undefined,
```

Comment:
This isn't 100% done because I've got an open question in the Puck discord about how to fully hide these components